Add io.github.hermitdemschoenenleben.linien

hermitdemschoenenleben opened the pull request:
Linien is a scientific lab software that allows for stabilizing the frequency of lasers.

bilelmoussaoui commented:
bot, build io.github.hermitdemschoenenleben.linien

LeandroStanger commented on io.github.hermitdemschoenenleben.linien.yml:
```diff
@@ -0,0 +1,112 @@
+app-id: io.github.hermitdemschoenenleben.linien
+runtime: org.kde.Platform
+runtime-version: '5.12'
```
Update runtime to 5.14
```suggestion
runtime-version: '5.14'
```

barthalion commented on io.github.hermitdemschoenenleben.linien.yml:
```diff
@@ -0,0 +1,112 @@
+  - python3-cython.json
+  - python3-wheel.json
+  - python3-numpy.json
+  - lapack.json
```
Any reason to keep lapack/boost in separate files? Besides, you can generate modules for python dependencies with https://github.com/flatpak/flatpak-builder-tools/tree/master/pip

hermitdemschoenenleben replied:
no, there was no reason for this, I just changed it

barthalion commented on io.github.hermitdemschoenenleben.linien.yml:
```diff
@@ -0,0 +1,112 @@
+sdk: org.kde.Sdk
+command: launcher.sh
+finish-args:
+  - --socket=x11
```
You want `--device=dri` for GPU acceleration. Not crucial but will improve GUI performance.

hermitdemschoenenleben replied:
ah, great! Thanks, I just added this option

barthalion commented on io.github.hermitdemschoenenleben.linien.metainfo.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
```
builddir/files/share/appdata/io.github.hermitdemschoenenleben.linien.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <description> required
```

hermitdemschoenenleben commented:
In the latest commit I
 - updated runtime to 5.14
 - added missing stuff to metainfo.xml
 - added --device=dri
 - combined lapack+scipy into a single file

bilelmoussaoui commented:
bot, build io.github.hermitdemschoenenleben.linien

hermitdemschoenenleben commented:
When trying the test build on different computers, I noticed some problems that I think are related to --device=dri. I just disabled dri in the last commit. Could you trigger a new build, please?

barthalion commented:
bot, build io.github.hermitdemschoenenleben.linien

hermitdemschoenenleben commented:
great, this build seems to work!

barthalion commented:
/merge
